notebook/static/notebook/js/notebook.js

```diff
@@ -2449,69 +2451,81 @@ define([
 
     /**
-     * Debug cells corresponding to the given indices.
-     *
-     * @param {Array} indices - indices of the cells to execute
+     * Reset code types set
      */
 
-    Notebook.prototype.any_debug_cell = function () {
-        var cells = this.get_cells();
-        for (var i = 0; i < cells.length; i++) {
-            if (cells[i].debug) {
-                return true;
-            }
-        }
-        return false;
+    Notebook.prototype.reset_code_types = function () {
+        var that = this;
+        this.code_types = ["main", "data", "debug"];
+
+        var code_types_submenu = $("#menu-code-type-submenu");
+        var hide_cells = $("#menu-hide-cells-submenu");
+        code_types_submenu.empty();
+        this.code_types.forEach(function (code_type) {
+            code_types_submenu.append(
+                $("<li>").attr("id", "code-type-submenu-" + code_type).append(
+                    $('<a>')
+                        .attr('href', '#')
+                        .click( function () {
+                            that.change_code_type(code_type);
+                        })
+                        .text(code_type)
+                )
+            );
+        });
+        hide_cells.empty();
+        this.code_types.forEach(function (code_type) {
+            hide_cells.append(
+                $("<li>").attr("id", "code-type-submenu-" + code_type).append(
+                    $('<a>')
+                        .attr('href', '#')
+                        .click( function () {
+                            that.hide_unhide_cells(code_type);
+                        })
+                        .text(code_type)
+                )
+            );
+        });
     }
 
-    Notebook.prototype.debug_cells = function (indices) {
+    /**
+     * Set cell types to cells corresponding to the given indices.
+     *
+     * @param {Array} indices - indices of the cells to execute
+     */
+
+    Notebook.prototype.change_code_types = function (indices, code_type) {
         if (indices.length === 0) {
             return;
         }
 
-        let all_debug_cell = true;
-
         var cell;
+
         for (var i = 0; i < indices.length; i++) {
             cell = this.get_cell(indices[i]);
-            if (cell.debug !== true) {
-                all_debug_cell = false;
-            }
+            cell.set_code_type(code_type);
         }
 
-        if (all_debug_cell) {
-            // If all cells are already in debug mode, then we need to
-            // turn off debug mode.
-            for (var i = 0; i < indices.length; i++) {
-                cell = this.get_cell(indices[i]);
-                cell.unmake_debug();
-            }
-        } else {
-            // Otherwise, we just need to make all of them a debug cell.
-            for (var i = 0; i < indices.length; i++) {
-                cell = this.get_cell(indices[i]);
-                cell.make_debug();
-            }
-        }
+        this.reset_code_types();
```

Comment:
Why call this here?

Reply:
I thought we might want to dynamically change the set of possible code types, if we remove all debug cells, then we won't need to have a button to hide debug cells.